src/lib.rs

```diff
@@ -314,6 +314,31 @@ where
             continue;
         }
 
+        // APEv2 frame
+        let is_ape_v2 = header_buffer[0] == 'A' as u8
+            && header_buffer[1] == 'P' as u8
+            && header_buffer[2] == 'E' as u8
+            && header_buffer[3] == 'T' as u8;
```

Comment:
Can you also check header_buffer[4] through header_buffer [7] for the full preamble (http://wiki.hydrogenaud.io/index.php?title=APE_Tags_Header)?

Reply:
header_buffer is only 4 bytes big,
I'll use ape_header instead and throw an UnexpectedFrame Error if it's wrong, would that be ok?

Reply:
That works for me!